src/Optimizely/Optimizely.php

```diff
@@ -388,4 +369,170 @@ public function getForcedVariation($experimentKey, $userId)
+            return null;
+        }
+
+        if(!$featureFlagKey){
+            $this->_logger->log(Logger::ERROR, "Feature Flag key cannot be empty");
+            return null;
+        }
+
+        if(!$userId){
+            $this->_logger->log(Logger::ERROR, "User ID cannot be empty");
+            return null;
+        }
+
+        $feature_flag = $this->_config->getFeatureFlagFromKey($featureFlagKey);
+        if($feature_flag == new FeatureFlag){
+            // Error logged in ProjectConfig - getFeatureFlagFromKey
+            return null;
+        }
+
+        $decision = $this->_decisionService->getVariationForFeature($feature_flag, $userId, $attributes);
+        if(!$decision){
+            $this->_logger->log(Logger::INFO,"Feature Flag '{$featureFlagKey}' is not enabled for user '{$userId}'.");
+            return false;
+        }
+
+        if($decision["experiment"]){
+            $experiment_key = $decision["experiment"]->getKey();
+            $variation_key = $decision["variation"]->getKey();
+
+            $this->sendImpression($experiment_key, $variation_key, $userId, $attributes);
+        } else {
+            $this->_logger->log(Logger::INFO,"The user '{$userId}' is not being experimented on Feature Flag '{$featureFlagKey}'.");
+        }
+
+        $this->_logger->log(Logger::INFO,"Feature Flag '{$featureFlagKey}' is enabled for user '{$userId}'.");  
+        return true;      
+    }
+
+    public function getFeatureVariableValueForType($featureFlagKey, $variableKey, $userId, $attributes, $variableType){
+
+        if(!$featureFlagKey){
+            $this->_logger->log(Logger::ERROR, "Feature Flag key cannot be empty");
+            return null;
+        }
+
+        if(!$variableKey){
+            $this->_logger->log(Logger::ERROR, "Variable key cannot be empty");
+            return null;
+        }
+
+        if(!$userId){
+            $this->_logger->log(Logger::ERROR, "User ID cannot be empty");
+            return null;
+        }
+
+        $feature_flag = $this->_config->getFeatureFlagFromKey($featureFlagKey);
+        if($feature_flag == new FeatureFlag){
+            // Error logged in ProjectConfig - getFeatureFlagFromKey
+            return null;
+        }
+
+        $variable = $this->_config->getFeatureVariableFromKey($featureFlagKey, $variableKey);
+        if(!$variable){
+            // Error message logged in ProjectConfig- getFeatureVariableFromKey
+            return null;
+        }
+
+        if($variableType != $variable->getType()){
+            $this->_logger->log(
+                Logger::ERROR,"Variable is of type {$variable->getType}, but you requested it as type {$variableType}");
+            return null;
+        }
+
+        $decision = $this->_decisionService->getVariationForFeature($feature_flag, $userId, $attributes);
+        $variable_value = $variable->getDefaultValue();
+
+        if(!$decision){
```

Comment:
first if decision then not

Reply:
I find this approach easier to read. Learned from Ruby code flow. 
The condition with lesser logic/immediate return should precede the one with more work